include/deal.II/base/complex_overloads.h

```diff
@@ -28,6 +28,14 @@ struct EnableIfScalar;
 template <typename T, typename U>
 struct ProductType;
 
+namespace Differentiation
```

Comment:
Please add `#ifndef DOXYGEN` when adding forward declaration :)
I have already added this preprocessor directive to this file in #8585. So it is probably helpful if you rebase this PR on the latest `upstream/master`.